plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -336,4 +338,13 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
                 .map(m -> new MongoColumnHandle(m.getName(), m.getType(), m.isHidden()))
                 .collect(toList());
     }
+
+    private static void validateColumnNamesForInsert(List<MongoColumnHandle> columns)
+    {
+        for (MongoColumnHandle column : columns) {
+            if (column.getName().contains("$") || column.getName().contains(".")) {
+                throw new IllegalArgumentException("Column name must not contain '$' or '.' for INSERT: " + column.getName());
+            }
+        }
+    }
```

Comment:
Let's change like below. I was missing MongoDB connector supports hidden column:
```java
    private static void validateColumnNameForInsert(MongoColumnHandle column)
    {
        if (column.getName().contains("$") || column.getName().contains(".")) {
            throw new IllegalArgumentException("Column name must not contain '$' or '.' for INSERT: " + column.getName());
        }
    }
```

Then, `beginInsert` method will be:
```java
        return new MongoInsertTableHandle(
                table.getSchemaTableName(),
                columns.stream()
                        .filter(column -> !column.isHidden())
                        .peek(MongoMetadata::validateColumnNameForInsert)
                        .collect(toImmutableList()));
```